core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -405,15 +415,12 @@ public void sendMessage(ReqT message) {
     try {
       // TODO(notcarl): Find out if messageIs needs to be closed.
       InputStream messageIs = method.streamRequest(message);
-      if (!useGet()) {
+      if (!useGetMethod) {
         stream.writeMessage(messageIs);
       } else {
         // Encode the request message and put it in the request headers.
-        byte[] payload = new byte[messageIs.available()];
-        messageIs.read(payload, 0, messageIs.available());
-        headers.put(
-            Metadata.Key.of("grpc-payload-bin", Metadata.BINARY_BYTE_MARSHALLER),
-            BaseEncoding.base64().encode(payload).getBytes());
+        byte[] payload = ByteStreams.toByteArray(messageIs);
```

Comment:
`ByteStreams` is beta. We can't use it except in tests. There is io.grpc.internal.IoUtils.toByteArray.